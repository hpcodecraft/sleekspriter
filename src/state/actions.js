export const brightnessToolIntensity = (intensity) => ({ type: "BRIGHTNESSTOOL_INTENSITY", intensity });
export const brightnessToolMode = (mode)  => ({ type: "BRIGHTNESSTOOL_MODE", mode });
export const boxFold = (box) => ({ type: "BOX_FOLD", box });
export const brushColor = (color) => ({ type: "BRUSH_COLOR", color });
export const frameSelect = (frame) => ({ type: "FRAME_SELECT", frame });
export const gridToggle = () => ({ type: "GRID_TOGGLE" });
export const layerAdd = (frame, layer) => ({ type: "LAYER_ADD", frame, layer });
export const layerDelete = (layer) => ({ type: "LAYER_DELETE", layer });
export const layerName = (layer, name) => ({ type: "LAYER_NAME", layer, name });
export const layerSelect = (layer) => ({ type: "LAYER_SELECT", layer });
export const layerOpacity = (layer, opacity) => ({ type: "LAYER_OPACITY", layer, opacity });
export const layerVisibility = (layer, visible) => ({ type: "LAYER_VISIBILITY", layer, visible });
export const modalHide = () => ({ type: "MODAL_HIDE" });
export const modalShow = (dialog) => ({ type: "MODAL_SHOW", dialog });
export const onionFrame = (mode, frame) => ({ type: "ONION_FRAME", mode, frame });
export const onionMode = (mode) => ({ type: "ONION_MODE", mode });
export const onionToggle = () => ({ type: "ONION_TOGGLE" });
export const paletteSelect = (palette) => ({ type: "PALETTE_SELECT", palette });
export const toolSelect = (tool) => ({ type: "TOOL_SELECT", tool });
export const zoomIn = () => ({ type: "ZOOM_IN" });
export const zoomOut = () => ({ type: "ZOOM_OUT" });
export const zoomSelect = (zoom) => ({ type: "ZOOM_SELECT", zoom });
export const zoomFit = (fileSize) => ({ type: "ZOOM_FIT", fileSize });
