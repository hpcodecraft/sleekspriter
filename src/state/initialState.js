const state = {
  ui: {
    paint: {
      tool: 'EraserTool',
      grid: true,
    },
  },
};

export default state;
