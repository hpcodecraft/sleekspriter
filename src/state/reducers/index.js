import uiReducer from './uiReducer';
import fileReducer from './fileReducer';

export {
  uiReducer,
  fileReducer,
};
