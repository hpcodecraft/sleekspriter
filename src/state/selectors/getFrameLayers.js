// returns an array containing the layers of selected painting frame
import { getPaintFrame, getFileLayers } from "./inputSelectors";

const getFrameLayers = state => {
  const frame = getPaintFrame(state);
  const layers = getFileLayers(state);
  return layers.filter(layer => layer.frame === frame);
};

export default getFrameLayers;
