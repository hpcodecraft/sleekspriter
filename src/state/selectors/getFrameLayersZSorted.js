// returns an array containing the layers of selected painting frame,
// sorted by z-index. (descending, 0 = lowest)
import { getFrameLayers } from ".";
import { zSorter } from "../../utils";

const getFrameLayersZSorted = state => {
  return getFrameLayers(state).sort(zSorter);
};

export default getFrameLayersZSorted;
