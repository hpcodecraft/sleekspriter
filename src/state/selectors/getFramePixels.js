// returns an object containing the pixels of selected painting frame
import { getPaintFrame, getFilePixels } from ".";

const getFramePixels = state => {
  const frame = getPaintFrame(state);
  const pixels = getFilePixels(state);
  return pixels[frame];
};

export default getFramePixels;
