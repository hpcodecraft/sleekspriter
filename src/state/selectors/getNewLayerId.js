// returns the total number of frames in the file
import { maxBy } from "lodash";
import { getFileLayers } from ".";

const getNewLayerId = state => {
  const layers = getFileLayers(state);
  return maxBy(layers, layer => layer.id).id + 1;
};

export default getNewLayerId;
