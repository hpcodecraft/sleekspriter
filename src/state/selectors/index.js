export { default as getFrameLayers } from "./getFrameLayers";
export { default as getFrameLayersZSorted } from "./getFrameLayersZSorted";
export { default as getFramePixels } from "./getFramePixels";
export { default as getNewLayerId } from "./getNewLayerId";
export { default as getOnionFrameAbsolute } from "./getOnionFrameAbsolute";
export { default as getOnionFrameLayersZSorted } from "./getOnionFrameLayersZSorted";
export { default as getPaintLayer } from "./getPaintLayer";
export { default as getTotalFrames } from "./getTotalFrames";
export * from "./stateSelectors";
