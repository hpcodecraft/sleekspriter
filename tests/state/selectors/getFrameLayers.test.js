import state from "../../__mocks__/state";
import { getFrameLayers } from "state/selectors";

describe("getFrameLayers", () => {
  it("returns all layers of a given frame", () => {
    expect(getFrameLayers(state)).toMatchSnapshot();
  });
});
