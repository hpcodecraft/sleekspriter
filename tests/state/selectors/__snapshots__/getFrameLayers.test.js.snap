// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFrameLayers returns all layers of a given frame 1`] = `
Array [
  Object {
    "frame": 1,
    "id": 2,
    "name": "Layer 2",
    "opacity": 100,
    "visible": true,
    "z": 1,
  },
  Object {
    "frame": 1,
    "id": 1,
    "name": "Layer 1",
    "opacity": 100,
    "visible": true,
    "z": 0,
  },
  Object {
    "frame": 1,
    "id": 3,
    "name": "Layer 3",
    "opacity": 100,
    "visible": true,
    "z": 2,
  },
]
`;
